core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java

```diff
@@ -174,6 +175,21 @@ private void acquireConnectionAndExecute(ClientRequestContext ctx, Endpoint endp
                                              HttpRequest req, DecodedHttpResponse res,
                                              ClientConnectionTimingsBuilder timingsBuilder,
                                              ProxyConfig proxyConfig) {
+        final InetSocketAddress remoteAddress = endpoint.toSocketAddress(-1);
+        try {
+            final boolean isValidIpAddr = factory.options().ipAddressFilter().test(remoteAddress);
+            if (!isValidIpAddr) {
+                final IpAddressRejectedException cause = new IpAddressRejectedException(
+                        "Invalid IP address: " + remoteAddress + " (endpoint: " + endpoint + ')');
+                earlyFailedResponse(cause, ctx);
+                return;
+            }
+        } catch (Throwable t) {
+            earlyFailedResponse(new IllegalStateException(
+                    "Unexpected exception from " + factory.options().ipAddressFilter(), t), ctx);
+            return;
```

Comment:
Note: If dns timings were recorded via `timingsBuilder`, they won't be recorded.

Reply:
That's correct. I think we should record the timing. @ikhoon Could you address this comment?

Reply:
`ClientConnectionTimingsBuilder` is designed to log with `Channel`, `SessionProcotol`.
https://github.com/line/armeria/blob/79112f537eb851aa2aba3b534777a608f2c71e0d/core/src/main/java/com/linecorp/armeria/common/logging/RequestLogBuilder.java#L81-L82
If we need to log `timingsBuilder` to `RequestLog`, the API signature should be fixed. Would you like to make the changes in this direction?

Previously, if an error occurred before acquiring the connection, `timingsBuilder` was not recorded.
https://github.com/line/armeria/blob/ad577f405c1f6e69e9f43a19982420a0876bfdf9/core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java#L202-L213

Reply:
It records with the null channel if DNS resolution was happened. 😉 
https://github.com/line/armeria/blob/fd540d449136f91e9b77b5ceb9d8ba7cf154cfea/core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java#L136